### PR TITLE
support nerdctl

### PR DESCRIPTION
**Pull request by ningmingxiao**
Signed-off-by: ningmingxiao <ning.mingxiao@zte.com.cn>

### CODE REVIEW

**Comment by ningmingxiao** [comment]
@AkihiroSuda

**Comment by AkihiroSuda** [comment]
@ningmingxiao Please explain what was the issue you were seeing, how you tested this PR, and how the original issue was solved

File: container.fc [hunk 1]
```diff
@@ -70,6 +70,7 @@
 /var/lib/containerd(/.*)?	gen_context(system_u:object_r:container_var_lib_t,s0)
 /var/lib/containerd/[^/]*/snapshots(/.*)?	gen_context(system_u:object_r:container_ro_file_t,s0)
 /var/lib/containerd/[^/]*/sandboxes(/.*)?	gen_context(system_u:object_r:container_ro_file_t,s0)
+/var/lib/nerdctl(/.*)? gen_context(system_u:object_r:container_ro_file_t,s0)
```
Comment (AkihiroSuda) [inline container.fc:73]: How does this work with writable volumes?

**Comment by ningmingxiao** [comment]
> @ningmingxiao Please explain what was the issue you were seeing, how you tested this PR, and how the original issue was solved

no issue, /var/lib/nerdctl  doesn't have selinux label, so I think it is necessary to add selinux lable to make it more safely

**Comment by AkihiroSuda** [comment]
> > @ningmingxiao Please explain what was the issue you were seeing, how you tested this PR, and how the original issue was solved
> 
> no issue, /var/lib/nerdctl doesn't have selinux label, so I think it is necessary to add selinux lable to make it more safely

"no issue" contradicts "necessary".

File: container.fc (hunk 1 above)
Comment (ningmingxiao) [reply container.fc:73]: I add  /var/lib/nerdctl/[^/]*/volumes/[^/]*/.* gen_context(system_u:object_r:container_file_t,s0)

Comment (AkihiroSuda) [reply container.fc:73]: How did you test this?

File: container.fc [hunk 2]
```diff
@@ -70,6 +70,8 @@
...
+/var/lib/nerdctl/[^/]*/volumes/[^/]*/.*	gen_context(system_u:object_r:container_file_t,s0)
```
[the remark this answers is not in the document]
Comment (ningmingxiao) [reply container.fc]: /var/lib/nerdctl is created by nerdctl not containerd, I want to know   if i don't add
```
/usr/s?bin/nerdctl.*		--	gen_context(system_u:object_r:container_runtime_exec_t,s0)
```
 are these  label will be  effective ？

```
/var/lib/nerdctl(/.*)? gen_context(system_u:object_r:container_ro_file_t,s0)
/var/lib/nerdctl/[^/]*/volumes/[^/]*/.*	gen_context(system_u:object_r:container_file_t,s0)
```

[the remark this answers is not in the document]
Comment (rhatdan) [reply container.fc]: Yes the labels on the /var/lib directories have nothing to do with the label of /usr/bin/nerdctl

LGTM

**Comment by rhatdan** [comment]
If you apply this policy make sure 

$ matchpathcon /var/lib/nerdctl
$ matchpathcon /var/lib/nerdctl/foobar
$ matchpathcon /var/lib/nerdctl/abcd/volumes
$ matchpathcon /var/lib/nerdctl/abcd/volumes/foobar
give you the correct labels.

You can also run `restorecon -R -v  /var/lib/nerdctl `

**Comment by rhatdan** [comment]
LGTM

**Comment by AkihiroSuda** [comment]
I’m still unsure and worried about how this PR was verified with nerdctl.
I also don’t see a PR (or a forked branch) to implement labeling on nerdctl side.

**Comment by ningmingxiao** [comment]
> > @ningmingxiao Please explain what was the issue you were seeing, how you tested this PR, and how the original issue was solved
> 
> no issue, /var/lib/nerdctl doesn't have selinux label, so I think it is necessary to add selinux lable to make it more safely

I use this demo  https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/using_selinux/writing-a-custom-selinux-policy_using-selinux

change /var/log/messages to 
/var/lib/nerdctl/1935db59/containers/default/0e626990a048a21b26b4cd5e3a2c32c05161ae21f87e783ba37103acc8855ecd/resolv.conf
can see avc:  denied  { append } for  pid=93549 comm="mydaemon" name="resolv.conf"

**Pull request merged**